### PR TITLE
add an 3 sec timeout for cut processing

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/CutDB.hs [hunk 1]
```diff
@@ -590,24 +591,28 @@ cutHashesToBlockHeaderMap
...
+
+            hdrs <- emptyCas
+            casInsertBatch hdrs $ V.fromList $ HM.elems $ _cutHashesHeaders hs
+
+            (headers :> missing) <- S.each (HM.toList $ _cutHashes hs)
+                & S.map (fmap snd)
+                & S.mapM (tryGetBlockHeader hdrs plds)
+                & S.partitionEithers
+                & S.fold_ (\x (cid, h) -> HM.insert cid h x) mempty id
+                & S.fold (\x (cid, h) -> HM.insert cid h x) mempty id
```
Comment (emilypi) [inline src/Chainweb/CutDB.hs]: why these back-to-back folds?

Comment (larskuhtz) [reply src/Chainweb/CutDB.hs]: this is unrelated to the PR. Let's take this offline.

File: src/Chainweb/CutDB.hs [hunk 2]
```diff
@@ -590,24 +591,28 @@ cutHashesToBlockHeaderMap
...
-        (headers :> missing) <- S.each (HM.toList $ _cutHashes hs)
-            & S.map (fmap snd)
-            & S.mapM (tryGetBlockHeader hdrs plds)
-            & S.partitionEithers
-            & S.fold_ (\x (cid, h) -> HM.insert cid h x) mempty id
-            & S.fold (\x (cid, h) -> HM.insert cid h x) mempty id
-        if null missing
-            then return $! Right headers
-            else return $! Left missing
+cutHashesToBlockHeaderMap logfun headerStore payloadStore hs = timeout 1000000 go >>= \case
```
Comment (larskuhtz) [inline src/Chainweb/CutDB.hs]:
```suggestion
cutHashesToBlockHeaderMap logfun headerStore payloadStore hs = timeout 3000000 go >>= \case
```

**Comment by slpopejoy** [comment]
Merged into #635

**Pull request closed without merging**